### PR TITLE
Fix issues identified during API review

### PR DESCRIPTION
**Pull request by luksa**

### CODE REVIEW

**Review by nrfox — COMMENTED** [review]
LGTM all issues I found are fixed

File: chart/crds/operator.istio.io_istiocnis.yaml [hunk 1]
```diff
@@ -1039,6 +1039,10 @@ spec:
                           enabled:
                             description: Controls whether ambient redirection is enabled
                             type: boolean
+                          ipv6:
```
Comment (dgn) [inline chart/crds/operator.istio.io_istiocnis.yaml:1042]: hmm. as we don't support ambient yet, maybe we want to remove this for now?

**Review by dgn — APPROVED** [review]
lgtm except for one question. Adding hold in case you want to make further changes, feel free to remove

/hold

**Comment by dgn** [comment]
just noticed that this still contains the `autoscalingv2` field: https://github.com/istio-ecosystem/sail-operator/blob/ac99ae0c4612444ae7feaaea7c73ff31cb2d3bff/api/v1alpha1/values_types.gen.go#L410-L412

I can submit a patch to remove it upstream and then update the commit we use in a separate PR

File: chart/crds/operator.istio.io_istiocnis.yaml (hunk 1 above)
Comment (luksa) [reply chart/crds/operator.istio.io_istiocnis.yaml:1042]: Removed. Thanks.

**Comment by luksa** [comment]
> just noticed that this still contains the `autoscalingv2` field:
> I can submit a patch to remove it upstream and then update the commit we use in a separate PR

Go ahead and remove it from upstream. But you don't need to update the commit, because I'm working on making the API transformer use the commits in go.mod, so your changes will automatically be pulled in when istio is updated.

**Pull request merged**